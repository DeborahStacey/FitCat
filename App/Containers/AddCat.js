import React from 'react'
import { ListView, View, ScrollView, Text, TouchableHighlight } from 'react-native'
import { Actions as NavigationActions } from 'react-native-router-flux'
import I18n from 'react-native-i18n'
import { Colors } from '../Themes/'
import RoundedButton from '../Components/RoundedButton'

import { default as WellCatManager } from '../Services/WellCatManager'

import styles from './Styles/AddCatStyle'

export default class AddCat extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      dataSource: new ListView.DataSource({
        rowHasChanged: (row1, row2) => row1 !== row2,
        sectionHeaderHasChanged: (h1, h2) => h1 !== h2
      }),
      catsList: {},
      loaded: false
    }

    this.fetchCatsList()
  }

  async fetchCatsList () {
    var catsList = await WellCatManager.fetchCatsList()
    console.log(catsList)
    var ds = this.state.dataSource.cloneWithRowsAndSections(catsList)

    this.setState({
      catsList: ds,
      loaded: true
    })
  }

  addCat (cat) {
    WellCatManager.addCat(cat)
    NavigationActions.dashboard()
  }

  renderSectionHeader (sectionData, sectionID) {
    return (
      <View style={styles.header}>
        <Text style={styles.headerText}>{sectionID}</Text>
      </View>
    )
  }

  renderRow (cat) {
    return (
      <TouchableHighlight underlayColor={Colors.frost} onPress={() => { this.addCat(cat) }}>
        <View style={styles.catRow}>
          <Text style={styles.catRowText}>{cat.name}</Text>
        </View>
      </TouchableHighlight>
    )
  }

  renderSeparator (sectionID, rowID, adjacentRowHighlighted) {
    return (
      <View
        key={`${sectionID}-${rowID}`}
      />
    )
  }

  render () {
    return (
      <View style={styles.mainContainer}>
        <ScrollView style={styles.container}>
          {
            this.state.loaded &&
            <ListView dataSource={this.state.catsList}
              renderSectionHeader={this.renderSectionHeader}
              renderRow={this.renderRow.bind(this)}
              renderSeparator={this.renderSeparator}
              style={styles.listView} />
          }
        </ScrollView>
        <View style={styles.paddedContainer}>
          <RoundedButton onPress={NavigationActions.newCat}>
            {I18n.t('new_cat')}
          </RoundedButton>
        </View>
      </View>
    )
  }
}
