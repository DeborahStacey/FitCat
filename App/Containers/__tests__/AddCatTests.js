import React from 'react'
import AddCat from '../AddCat'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('renders the add cat page corectly', () => {
  const tree = renderer.create(
    <AddCat />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
