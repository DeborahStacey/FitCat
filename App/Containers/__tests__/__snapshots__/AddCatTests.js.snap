exports[`test renders the add cat page corectly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F1F5FF",
      "flex": 1,
      "marginTop": 64,
    }
  }>
  <ScrollView
    style={
      Object {
        "flex": 1,
        "height": 500.25,
        "paddingTop": 10,
      }
    } />
  <View
    style={
      Object {
        "flex": 1,
        "paddingBottom": 50,
        "paddingTop": 10,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#466ECC",
          "borderRadius": 5,
          "height": 45,
          "justifyContent": "center",
          "marginHorizontal": 25,
          "marginVertical": 10,
          "opacity": 1,
        }
      }
      testID={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 5,
            "marginVertical": 10,
            "textAlign": "center",
          }
        }>
        [MISSING \"EN.NEW_CAT\" TRANSLATION]
      </Text>
    </View>
  </View>
</View>
`;
